sdb/commands/member.py

```diff
@@ -17,13 +17,79 @@
 # pylint: disable=missing-docstring
 
 import argparse
-from typing import Iterable
+from enum import Enum
+from typing import Iterable, List, Tuple
+import re
 
 import drgn
 import sdb
 
 
+class MemberExprSep(Enum):
+    START = "__start__"
+    DOT = "."
+    PTR = "->"
+    ARRAY = "[]"
+
+
 class Member(sdb.Command):
+    """
+    Dereference members of the given structure
+
+    DESCRIPTION
+        This is one of main commands used in SDB to explore members of C
+        structures and traverse them hierarchically when these structures are
+        nested. The notation used for this command is best demonstrated in the
+        EXAMPLES section below but in general users should expect the notation
+        used in standard C:
+        [1] The dot notation/operator `.` can be used reference members in an
+            embedded struct.
+        [2] `->` can be used to walk into/dereference struct members that are
+            pointers.
+        [3] Array notation can be used to look into a specific element of an
+            array using the index notation. This includes both formally
+            specified arrays (like char[100]) and pointers that are used as
+            arrays allocated at runtime.
+
+        Something worth noting from a usability perspective for struct members
+        that are pointers (see [2] above) is that SDB relaxes the notation and
+        allows dereference with the `.` operator too. This means that something
+        like this:
+            `... | member ptr0->ptr1->ptr2`
+        could be also written like this:
+            `... | member ptr0.ptr1.ptr2`
+
+    EXAMPLES
```

Comment:
Looking at these examples, to help a naive user better understand what's being shown, it might be useful to add some text explanation for each one. E.g. add a quick header to explain what feature(s) are being shown off for each example, and perhaps why the feature(s) is being highlighted:
```

Extract "active_mm" member from "task_struct" structure:

        sdb> addr init_task | member active_mm
        *(struct mm_struct *)0xffff9d0ac59198c0 = {
                .mmap = (struct vm_area_struct *)0xffff9d0ac47179c0,

Extract non-existent member:


        sdb> addr init_task | member active_mm->hmm
        (struct hmm *)0x0
        sdb> addr init_task | member active_mm.hmm
        (struct hmm *)0x0

Extract first element from array:

        sdb> addr init_task | member active_mm.owner.vmacache.vmas[1]
        *(struct vm_area_struct *)0xffff9d0ac4717a90 = {
                .vm_start = (unsigned long)140161947738112,
        ...
```

I'm fine landing this as-is, but I think if I wasn't already familiar with this command, I think it might be a little difficult to parse these examples, and fully understand what is being shown.

Reply:
yeah that'd be helpful, working on it now.